tests/core/test_emscripten_api.in

```diff
@@ -27,7 +27,7 @@ int main() {
   buffer[1] = 2;
   buffer[2] = 3;
   int n = emscripten_print_double(d, buffer, 2);
-  assert(n == len);
+  assert(n == 1);
```

Comment:
Here the old implementation of `emscripten_print_double` was not actually behaving like it is documented. The documentation about the return value says "The number of necessary bytes, not including the null terminator (actually written, if ``to`` is not NULL)". When passing in a buffer of two bytes to write, the function can only write one character and then it has to write the null byte, so it should return 1 instead.

In most cases, I like the pattern where functions take in the number of bytes they can write, including null terminator, and return the number of bytes written, excluding null terminator, because this ensures that if one errs off by one, it will always be to the safe side, and never results in reading or writing past the bounds. This is how `emscripten_print_double` and `stringToUTF8` work.